### PR TITLE
add strtobool function in utils

### PR DESCRIPTION
**Pull request by odrling**
This PR is required to support Python 3.12.

Python 3.12 removed distutils and this utility function is not included in the setuptools package.
[PEP 632](https://peps.python.org/pep-0632/) suggests that strtobool should be reimplemented rather than relying on a compatibility package (unlike the rest of distutils).

This reimplementation doesn't accept all the inputs that distutils used to accept but it is should work fine for our usage.

### CODE REVIEW

File: src/dakara_base/config.py [hunk 1]
```diff
@@ -35,14 +35,15 @@
...
 from collections import UserDict
-from distutils.util import strtobool
 
 import coloredlogs
 import progressbar
 import yaml
 from environs import Env, EnvError
 from path import Path
 
+from .utils import strtobool
```
Comment (Neraste) [inline src/dakara_base/config.py]: Please replace by:

```python
from dakara_base.utils import strtobool
```

File: src/dakara_base/utils.py [hunk 2]
```diff
@@ -129,3 +129,18 @@ def create_url(
 
 class URLParameterError(DakaraError, ValueError):
     """Error raised when server parameters are unproperly set."""
+
+
+def strtobool(input_value: str, default: bool = False) -> bool:
+    """
+    Convert user input to a boolean
+    """
```
Comment (Neraste) [inline src/dakara_base/utils.py]: Please format the docstring as the others in the project (using the [Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)).

**Review by Neraste — CHANGES_REQUESTED** [review]

File: src/dakara_base/utils.py (hunk 2 above)
Comment (odrling) [reply src/dakara_base/utils.py]: should be fine with 35d2f690ca4efd918bda4fad67dddc2e6fa7fd7e

File: src/dakara_base/utils.py [hunk 3]
```diff
@@ -129,3 +129,26 @@ def create_url(
...
+
+
+def strtobool(input_value: str, default: bool = False) -> bool:
+    """Convert a string representation of truth to true or false.
+
+    True values are `y`, `yes`; false values are `n`, `no`.
+    Returns `default` if `input_value` is anything else.
+
+    Args:
+        input_value (str): string representing a boolean
```
Comment (Neraste) [inline src/dakara_base/utils.py:141]: Missing the default value description here.

File: tests/test_utils.py [hunk 4]
```diff
@@ -92,3 +92,18 @@ def test_invalid_furl(self, mocked_furl):
...
             create_url(host="www.example.com", scheme_no_ssl="http")
+
+
+class StrtoboolTests(TestCase):
+    def test_y(self):
+        """Test that y returns True"""
+        for uinput in ("y", "yes", "Y", "YES"):
+            self.assertTrue(strtobool(uinput))
+
+    def test_n(self):
```
Comment (Neraste) [inline tests/test_utils.py:103]: Missing description here.

File: tests/test_utils.py [hunk 5]
```diff
@@ -92,3 +92,18 @@ def test_invalid_furl(self, mocked_furl):
...
+        for uinput in ("n", "no", "N", "NO"):
+            self.assertFalse(strtobool(uinput, default=True))
+
+    def test_default(self):
```
Comment (Neraste) [inline tests/test_utils.py:110]: Missing description here too.

**Review by Neraste — APPROVED** [review]

**Pull request merged**